examples/fermat/twosq/cornerScript.sml

```diff
@@ -0,0 +1,1075 @@
+           ==> CARD s = 1                      by prime_corners_conjugate_fixes, CARD_SING
+            or ODD (CARD s)                    by ODD_1
+           Now FINITE (corners p)              by corners_finite
+           and conjugate involute (corners p)  by conjugate_involute_corners
+           and transpose involute (corners p)  by transpose_involute_corners
+           ==> ODD (CARD t)                    by involute_two_fixes_both_odd, [1]
+
+   Claim: EVEN (CARD w)
+   Proof: Note ?k. p = 4 * k + 1               by tik_exists
+           and CARD w = HALF (p - 1)           by prime_corners_eq_fix_card, prime p
+                      = HALF (4 * k)           by above
+                      = HALF (2 * (2 * k))     by arithmetic
+                      = 2 * k                  by HALF_TWICE
+          ==> EVEN (CARD w)                    by EVEN_DOUBLE, [2]
+
+   Note CARD u = 0                             by assumption, u = {}, CARD_EMPTY
+    and CARD v = 0                             by prime_corners_more_fix, CARD_EMPTY
+    Now CARD t = CARD u + CARD v + CARD w      by corners_transpose_fixes_card
+               = CARD w                        by above
+    But ODD (CARD t) /\ EVEN (CARD w)          by [1], [2]
+   This is a contradiction                     by ODD_EVEN
+*)
+Theorem tik_prime_corners_less_fix:
+  !p. tik p /\ prime p ==> corners_less_fix p <> {}
+Proof
+  rpt strip_tac >>
+  qabbrev_tac `s = fixes conjugate (corners p)` >>
+  qabbrev_tac `t = fixes transpose (corners p)` >>
+  qabbrev_tac `u = corners_less_fix p` >>
+  qabbrev_tac `v = corners_more_fix p` >>
+  qabbrev_tac `w = corners_eq_fix p` >>
+  `ODD (CARD t)` by
+  (`p <> 2` by metis_tac[odd_tik_tok, EVEN_2, ODD_EVEN] >>
+  `CARD s = 1` by fs[prime_corners_conjugate_fixes, Abbr`s`] >>
+  `ODD (CARD s)` by simp[ODD_1] >>
+  `FINITE (corners p)` by simp[corners_finite] >>
+  `conjugate involute (corners p)` by metis_tac[conjugate_involute_corners] >>
+  `transpose involute (corners p)` by metis_tac[transpose_involute_corners] >>
+  metis_tac[involute_two_fixes_both_odd]) >>
+  `EVEN (CARD w)` by
+    (`?k. p = 4 * k + 1` by simp[tik_exists] >>
+  `p - 1 = 2 * (2 * k)` by decide_tac >>
+  `HALF (p - 1) = 2 * k` by metis_tac[HALF_TWICE] >>
+  fs[prime_corners_eq_fix_card, EVEN_DOUBLE, Abbr`w`]) >>
+  `CARD u = 0` by fs[] >>
+  `CARD v = 0` by simp[prime_corners_more_fix, Abbr`v`] >>
+  metis_tac[corners_transpose_fixes_card, ADD, ODD_EVEN]
+QED
+
+(* Theorem: tik p /\ prime p ==> ?(a, b). p = a ** 2 + b ** 2 /\ a < b *)
+(* Proof:
+   Let s = corners_less_fix p.
+   Note s <> {}                                by tik_prime_corners_less_fix
+    ==> ?q. q IN s                             by MEMBER_NOT_EMPTY
+    ==> ?a b. q = (a,a,b,b) /\
+              p = a ** 2 + b ** 2 /\ a < b     by corners_less_fix_def
+   Take them and form a pair (a,b).
+*)
+Theorem fermat_two_squares_by_corners:
+  !p. tik p /\ prime p ==> ?(a, b). p = a ** 2 + b ** 2 /\ a < b
```

Comment:
Unless there's strong reason to, please don't use paired existentials, like `?(a,b)....` Instead, just have the existentially quantified variables listed: `?a b. ...`

Reply:
New commit uses simple existentially quantified variables. Should I worry about the failure of [d64379b]?

Reply:
Got it. Now I learn how to read the log, and fixed the problem. Hope this is the only one.

Reply:
This time it is the standard kernel, but I have used rename1 to ensure proper variable rename. Log seems not to fail in fermat/ folder. Any suggestion about the exact issue?

Reply:
Thanks for all your help, much appreciated.